lib/diagnostic.js

```diff
@@ -6,23 +6,48 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizeWords = function (string) {
+  const words = string.split(/\s+/)
+  const normalizedWords = []
+  for (let i = 0; i < words.length; i++) {
+    const word = words[i].replace(/\W+/, '')
+    normalizedWords[i] = word.toUpperCase()
+  }
+  return normalizedWords()
 }
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function (string)
+let uniqueWordsAsKeys = {};
+for (let i = 0; i < normalizedWords.length; i++) {
+  let word = normalizedWords[i]
+  uniqueWordsAsKeys[word] = true
+}
+
+// The keys pulled out into a list
+let uniqueWords = []
+
+let keys = [];
+let k = 0;
+for (keys[k++] in uniqueWordsAsKeys);
+
+uniqueWords = keys;
 }
 
 // 3. Write a function that takes a string argument and returns the count of
 // words in a string.
 // Provide the *option* to count unique words instead of total words, by passing
 // a second argument.
 
-const wordCount = function (/* */) {
-  /* your response here */
+const wordCount = function (string) {
+  let wordTotal = 0
+  for (let i = 0; i < string.length; i++) {
+    if (string.split[i] === ' ') {
```

Comment:
I think you actually want to split up the string by a space character, rather than `[i]`. If you split the string on `[i]`, you are asking to create a new item in an array every time the interpreter encounters `i`, which will be a number